### ISSUE TITLE
assert-on-string-literal message is wrong for empty literals 

### ISSUE THREAD

**Issue by mattwala** [PyCQA/pylint#3400]
### Steps to reproduce
1. Install pylint from master
2. Run pylint on the following code:
```python
assert ""
```

### Current behavior
```
$ pylint test.py
test.py:1:0: W0129: Assert statement has a string literal as its first argument. The assert will never fail. (assert-on-string-literal)
```

### Expected behavior

This assert will fail, so the message should not state the contrary.

It's not clear to me if the message should be changed, or if assert-on-string-literal should be changed ignore empty strings.

### pylint --version output
```
pylint 2.5.0-dev1
astroid 2.4.0
Python 3.7.3 | packaged by conda-forge | (default, Jul  1 2019, 14:38:56) 
[Clang 4.0.1 (tags/RELEASE_401/final)]
```
